connect/service.go

```diff
@@ -53,29 +53,20 @@ type Service struct {
 	logger hclog.Logger
 }
 
-// NewService creates and starts a Service. The caller must close the returned
-// service to free resources and allow the program to exit normally. This is
-// typically called in a signal handler.
-//
-// Caller must provide client which is already configured to speak to the local
-// Consul agent, and with an ACL token that has `service:write` privileges for
-// the service specified.
-func NewService(serviceName string, client *api.Client) (*Service, error) {
-	logger := hclog.New(&hclog.LoggerOptions{})
-
-	return NewServiceWithLogger(serviceName, client,
-		logger)
+type Config struct {
+	Client           *api.Client
+	Logger           hclog.Logger
+	ServerNextProtos *[]string
```

Comment:
Not sure if you planned to do it later but adding doc comments for a public Config API would be good. Most are trivial, but worth noting that Client is mandatory and will cause a panic if not set while logger is optional and default will apply etc.

ServerNextProtos could link to the discucssion in the issue for more details on what to set there perhaps?